test/test_interpreter.cpp

```diff
@@ -9,6 +9,9 @@
 #include "doctest/doctest.h"
 #include "xeus-cpp/xinterpreter.hpp"
 
+// Forward declaration of extract_filename
+std::string extract_filename(int &argc, char* argv[]);
```

Comment:
Now we need to call this function with different arguments and make sure it does what’s expected.

Reply:
Okay I'll add a basic test from what comes to my mind.